### PR TITLE
Added pomodoro funcionality

### LINKED ISSUE #3: Pomodoro
Implement Pomodoro timer in functions.c.

Inspiration of what [Pomodoro](https://pomofocus.io/) should do.

### PR DESCRIPTION
**Pull request by glklimmer**
Pomodoro now starts a thread that updates every second and counts down
the time until the pomodoro is over. Then rest starts.

Also just copied code for directory, needs refactor maybe considering ip is not shown for example.

Currently typing is overwritten by the thread, which means there needs
to be a way to save the typing. Will do this with arrow key feature.

### CODE REVIEW

**Comment by glklimmer** [comment]
@edizeqiri I just copied the code for console output, can we do this better?

**Comment by edizeqiri** [comment]
We could do a general function in `functions.c` for the console output and call it in `main` and `pomodoro`.

**Comment by edizeqiri** [comment]
> Currently typing is overwritten by the thread, which means there needs to be a way to save the typing. Will do this with arrow key feature.

We could update the line above and not the line under it which would fix this issue like this stackoverflow [thread](https://stackoverflow.com/questions/15192441/update-printf-value-on-same-line-instead-of-new-one).

**Comment by edizeqiri** [comment]
> Currently typing is overwritten by the thread, which means there needs to be a way to save the typing. Will do this with arrow key feature.

How about we only update the timer every minute or after pressing enter? Call this passive mode and active mode could be without a thread but as main process.

File: functions.c [hunk 1]
```diff
@@ -135,26 +136,86 @@ int unimode(int learning)
...
+    printf("To start The ZEN ᕙ(⇀‸↼)ᕗ again just type the command 'unimode'.\n");
   }
 
   return 1;
 }
 
-void pomodoro(int learn_int, int pause_int)
+void printWithTime(int h, int m, int s, int pause, int i, int interval)
 {
+  char cwd[PATH_MAX];
```
Comment (edizeqiri) [inline functions.c:147]: If you want we can do a function for the console output, but since time is spare we can merge this

**Review by edizeqiri — COMMENTED** [review]
Nice work. We can merge it, as it won't brake any functionality and is sufficient even with the little bugs.

If you like the idea of active/passive mode (my last comment) then we can implement it and merge. Else we can merge it now. Call is on you.

**Pull request merged**